nextest-metadata/src/test_list.rs

```diff
@@ -136,6 +133,28 @@ pub struct RustTestSuiteSummary {
 
     /// The path to the test binary executable.
     pub binary_path: Utf8PathBuf,
+}
+
+/// A serializable suite of Rust test binaries.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct RustBinaryListSummary {
+    /// The list of test binaries
+    pub binaries: Vec<RustTestBinarySummary>,
+}
+
```

Comment:
This should be a top-level `BinaryListSummary` with a `rust-binaries` key which contains a `BTreeMap<String, RustTestBinarySummary>`.

Regarding adding cargo metadata to the list, yeah I can see why that's useful! That's a bit hard to do at the moment because guppy doesn't expose (or even directly store) the metadata it was built from. What you could do is take the metadata at https://github.com/nextest-rs/nextest/blob/5fa8a77fec3310765d5a8260e5047ba61c09f4d9/cargo-nextest/src/dispatch.rs#L416-L417 and save it for later. Only parse it as JSON and include it in the final output if an option `--include-cargo-metadata` is passed in (and only accept `--include-cargo-metadata` if the output is serializable).

On the test running side, only accept `--bin-list` if the cargo metadata is available.

Reply:
I did think of keeping the raw metadata output, but wasn't sure about it... I will test it that way.

`rust_binaries` indexed by binary name ?

Reply:
Done